support/types.js

```diff
@@ -238,10 +238,21 @@ exports.isDataView = isDataView;
 function isSharedArrayBufferToString(value) {
   return ObjectToString(value) === '[object SharedArrayBuffer]';
 }
-isSharedArrayBufferToString.working = (
-  typeof SharedArrayBuffer !== 'undefined' &&
-  isSharedArrayBufferToString(new SharedArrayBuffer())
-);
+// Avoid invoking SharedArrayBuffer constructor until required, then memoize
+Object.defineProperty(isSharedArrayBufferToString, 'working', {
+  get: (function() {
```

Comment:
I wrote this as a getter to keep the same API of `<function_name>.working` - however this could be re-written any number of ways.

Reply:
a getter limits backwards compatibility, and also makes things slow. it would be ideal to avoid them.